source/extensions/omni.isaac.lab/test/envs/test_scale_randomization.py

```diff
@@ -0,0 +1,343 @@
+
+    joint_pos = CubeActionTermCfg(asset_name="cube1")
+
+
+@configclass
+class ObservationsCfg:
+    """Observation specifications for the MDP."""
+
+    @configclass
+    class PolicyCfg(ObsGroup):
+        """Observations for policy group."""
+
+        # cube velocity
+        position = ObsTerm(func=base_position, params={"asset_cfg": SceneEntityCfg("cube1")})
+
+        def __post_init__(self):
+            self.enable_corruption = True
+            self.concatenate_terms = True
+
+    # observation groups
+    policy: PolicyCfg = PolicyCfg()
+
+
+@configclass
+class EventCfg:
+    """Configuration for events."""
+
+    reset_base = EventTerm(
+        func=mdp.reset_root_state_uniform,
+        mode="reset",
+        params={
+            "pose_range": {"x": (-0.5, 0.5), "y": (-0.5, 0.5), "yaw": (-3.14, 3.14)},
+            "velocity_range": {
+                "x": (-0.5, 0.5),
+                "y": (-0.5, 0.5),
+                "z": (-0.5, 0.5),
+            },
+            "asset_cfg": SceneEntityCfg("cube1"),
+        },
+    )
+
+    # Scale randomization as intended
+    randomize_scale1 = EventTerm(
+        func=mdp.randomize_scale,
+        mode="scene",
+        params={
+            "scale_range": {"x": (0.5, 1.5), "y": (0.5, 1.5), "z": (0.5, 1.5)},
+            "asset_cfg": SceneEntityCfg("cube1"),
+        },
+    )
+
+    # Static scale values
+    randomize_scale2 = EventTerm(
+        func=mdp.randomize_scale,
+        mode="scene",
+        params={
+            "scale_range": {"x": (1.0, 1.0), "y": (1.0, 1.0), "z": (1.0, 1.0)},
+            "asset_cfg": SceneEntityCfg("cube2"),
+        },
+    )
+
+
+##
+# Environment configuration
+##
+
+
+@configclass
+class CubeEnvCfg(ManagerBasedEnvCfg):
+    """Configuration for the locomotion velocity-tracking environment."""
+
+    # Scene settings
+    scene: MySceneCfg = MySceneCfg(num_envs=10, env_spacing=2.5, replicate_physics=False)
+    # Basic settings
+    observations: ObservationsCfg = ObservationsCfg()
+    actions: ActionsCfg = ActionsCfg()
+    events: EventCfg = EventCfg()
+
+    def __post_init__(self):
+        """Post initialization."""
+        # general settings
+        self.decimation = 2
+        # simulation settings
+        self.sim.dt = 0.01
+        self.sim.physics_material = self.scene.terrain.physics_material
+
+
+class TestScaleRandomization(unittest.TestCase):
+    """Test for texture randomization"""
+
+    """
+    Tests
+    """
+
+    def test_scale_randomization(self):
+        """Main function."""
+
+        # setup base environment
+        env = ManagerBasedEnv(cfg=CubeEnvCfg())
+        # setup target position commands
+        target_position = torch.rand(env.num_envs, 3, device=env.device) * 2
+        target_position[:, 2] += 2.0
+        # offset all targets so that they move to the world origin
+        target_position -= env.scene.env_origins
+
+        stage = omni.usd.get_context().get_stage()
+
```

Comment:
Might make sense to also check that all assets in the scene are created successfuly. Sometimes PhysX might be erroring out which isn't desired.

Also you should add the failure case of replicate physics is not True and sim is playing so we can ensure we handle that.